apps/web-app/components/settings/addNewGroup.vue

```diff
@@ -4,47 +4,264 @@
-        groupColor: newGroupColorValue.value
+    if (createEmailIdentityForGroup.value) {
+      if (!newGroupColorValue.value) return;
+      buttonLoading.value = true;
+      buttonLabel.value = 'Creating...';
+      const toast = useToast();
+      const { newGroupPublicId } =
+        await $trpc.org.users.userGroups.createOrgUserGroups.mutate({
+          groupName: newGroupNameValue.value,
+          groupDescription: newGroupDescriptionValue.value,
+          groupColor: newGroupColorValue.value
+        });
+
+      const selectedGroupsPublicIds: string[] = selectedOrgGroups.value.map(
+        (group) => group.publicId as string
+      );
+      const selectedOrgMembersPublicIds: string[] =
+        selectedOrgMembers.value.map((member) => member.publicId as string);
+
+      await $trpc.org.mail.emailIdentities.createNewEmailIdentity.mutate({
+        emailUsername: newIdentityUsernameValue.value,
+        domainPublicId: selectedDomain.value?.domainPublicId as string,
+        sendName: newIdentitySendNameValue.value,
+        routeToGroupsPublicIds: [...selectedGroupsPublicIds, newGroupPublicId],
+        routeToUsersOrgMemberPublicIds: selectedOrgMembersPublicIds,
+        catchAll: newIdentityCatchAll.value
+      });
+      buttonLoading.value = false;
+      buttonLabel.value = 'Done... Redirecting';
+      toast.add({
+        id: 'address_added & group_created',
+        title: 'Address Added & Group Group Created',
+        description: `${newIdentityUsernameValue.value}@${selectedDomain.value?.domain} has been added successfully & Group has been created successfully`,
+        icon: 'i-ph-thumbs-up',
+        timeout: 5000
       });
-    buttonLoading.value = false;
-    buttonLabel.value = 'Done... Redirecting';
-    const toast = useToast();
-    toast.add({
-      id: 'group_xreated',
-      title: 'Group Group Created',
-      description: 'Group has been created successfully',
-      icon: 'i-ph-thumbs-up',
-      timeout: 5000
-    });
-    setTimeout(() => {
-      emit('close');
-    }, 1000);
+      setTimeout(() => {
+        emit('close');
+      }, 1000);
+    } else {
+      if (!newGroupColorValue.value) return;
+      buttonLoading.value = true;
+      buttonLabel.value = 'Creating...';
+      const { newGroupPublicId } =
+        await $trpc.org.users.userGroups.createOrgUserGroups.mutate({
+          groupName: newGroupNameValue.value,
+          groupDescription: newGroupDescriptionValue.value,
+          groupColor: newGroupColorValue.value
+        });
+      buttonLoading.value = false;
+      buttonLabel.value = 'Done... Redirecting';
+      const toast = useToast();
+      toast.add({
+        id: 'group_created',
+        title: 'Group Group Created',
+        description: 'Group has been created successfully',
+        icon: 'i-ph-thumbs-up',
+        timeout: 5000
+      });
+      setTimeout(() => {
+        emit('close');
+      }, 1000);
+    }
```

Comment:
if we move the group creation logic out of the if loop
this whole section can be removed